geetools/ee_image_collection.py

```diff
@@ -896,6 +896,25 @@ def groupInterval(self, unit: str = "month", duration: int = 1) -> ee.List:
             lambda dr: ic.filterDate(ee.DateRange(dr).start(), ee.DateRange(dr).end())
         )
 
+        sizeName = "__geetools_generated_size__"
+
+        def add_size(ic):
+            ic = ee.ImageCollection(ic)
+            return ic.set({sizeName: ic.size()})
+
+        def delete_size_property(ic):
+            ic = ee.ImageCollection(ic)
+            propertiesToCopy = ic.propertyNames().remove(sizeName)
```

Comment:
I assume that all the imagecollection will have the same property name so instead of recomputing it in each loop, you can extract it from the first imagecollection outsdide this method (e.g.) before `add_size`